lib/python/rose/macro.py

```diff
@@ -723,12 +725,18 @@ def pretty_format_config(config):
         for keylist, node in list(s_node.walk()):
             # FIXME: Surely, only the scheme know how to splits its array?
             values = rose.variable.array_split(node.value, ",")
-            node.value = pretty_format_value(values)   
+            node.value = pretty_format_value(values)
             new_keylist = pretty_format_keys(keylist)
-            if new_keylist != keylist:
+            if new_keylist != keylist and ignore_error is False:
                 s_node.unset(keylist)
                 s_node.set(new_keylist, node.value, node.state, node.comments)
-                
+                _report_error(text=ERROR_MACRO_CASE_MISMATCH.format(keylist[1],
```

Comment:
It's simpler if the `if` is around this part of the block and the sys.exit, I think.